test/basic/basic.go

```diff
@@ -187,3 +203,44 @@ func CreateInsecureConnection(targetAaddress string) (*grpc.ClientConn, error) {
 
 	return conn, nil
 }
+
+// ValidateHostDiscovery validates that all hosts get discovered
+func ValidateHostDiscovery(t *testing.T, id int) {
+	t.Logf("Creating gNMI connection for agent %d", id)
+	gconn, err := CreateInsecureConnection(fmt.Sprintf("discovery-agent-%d.discovery-agent:30000", id))
+	assert.NoError(t, err)
+	gnmiClient := gnmi.NewGNMIClient(gconn)
+
+	t.Logf("%d: Subscribing for gNMI notifications...", id)
+	ctx := context.Background()
+	subClient, err := gnmiClient.Subscribe(ctx)
+	assert.NoError(t, err)
+	err = subClient.Send(&gnmi.SubscribeRequest{
+		Request: &gnmi.SubscribeRequest_Subscribe{
+			Subscribe: &gnmi.SubscriptionList{
+				Subscription:     []*gnmi.Subscription{{Path: utils.ToPath("state/host[mac=...]")}},
+				Mode:             gnmi.SubscriptionList_STREAM,
+				AllowAggregation: true,
+			}},
+	})
+	assert.NoError(t, err)
+
+	// Wait until we get 8 host updates total
```

Comment:
I'd wait for 10 host updates (all hosts)

Reply:
Addressed